lti_provider/views.py

```diff
@@ -0,0 +1,114 @@
+import logging
+
+from django.core.urlresolvers import reverse
+from django.http import HttpResponse, Http404
+from django.shortcuts import redirect
+from django.views.decorators.csrf import csrf_exempt
+from lti.contrib.django import DjangoToolProvider
+from lti import ToolConfig
+from oauthlib import oauth1
+
+from backend.models import Assignment, Course
+from lti_provider import lti_settings as settings
+from lti_provider.models import Consumer, LTIUser
+from validator import RequestValidator
+
+
+logger = logging.getLogger(__name__)
+
+
+ROLES = {
+    'Instructor': 'instructor',
+    'Student': 'student',
+}
+
+
+def config(request):
+    # Code from lti_django example
+    # basic stuff
+    app_title = 'StarCellBio'
+    app_description = 'Star Cell Bio LTI Application'
+    launch_view_name = 'lti:launch_experiment'
+    launch_url = request.build_absolute_uri(reverse(launch_view_name, args=[
+        'course_id',
+        'assignment_id',
+        'experiment'
+    ]))
+
+    # maybe you've got some extensions
+    # FIXME(idegtiarov) there is no extentions right now, if they don't appear remove example.
+    extensions = {
+        'my_extensions_provider': {
+            # extension settings...
+        }
+    }
+
+    lti_tool_config = ToolConfig(
+        title=app_title,
+        launch_url=launch_url,
+        secure_launch_url=launch_url,
+        extensions=extensions,
+        description=app_description
+    )
+
+    return HttpResponse(lti_tool_config.to_xml(), content_type='text/xml')
+
+
+@csrf_exempt
+def lti_launch(request, course_id=None, assignment=None, experiment=None):
+    """
+    LTI main view
+
+    Analyze LTI POST request to launch LTI session
+
+    :param request: LTI request
+    :param course_id: assingment id from the launch URL
+    :param experiment: experiment id from the launch URL
+    """
+    request_post = request.POST
+
+    if settings.DEBUG_LTI:
+        logger.debug(request.META)
+        logger.debug(request_post)
+
+    try:
+        tool_provider = DjangoToolProvider.from_django_request(request=request)
+        validator = RequestValidator()
+        ok = tool_provider.is_valid_request(validator)
+    except oauth1.OAuth1Error as err:
```

Comment:
Here can be raised two other errors InvalidLTIRequestError abd ValueError. Add them in except checking block.